pkg/tarfs/tarfs.go

```diff
@@ -457,18 +457,38 @@ func (f *FS) ReadFile(name string) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	if i.h.FileInfo().Mode().Type()&fs.ModeSymlink != 0 {
+
+	dataSize := i.h.Size
+	typ := i.h.FileInfo().Mode().Type()
+	var r *tar.Reader
+	switch {
+	case typ.IsRegular() && i.h.Typeflag != tar.TypeLink:
+		r = tar.NewReader(io.NewSectionReader(f.r, i.off, i.sz))
+	case typ.IsRegular() && i.h.Typeflag == tar.TypeLink:
+		tgt, err := f.getInode(op, i.h.Linkname)
+		if err != nil {
+			return nil, err
+		}
+		r = tar.NewReader(io.NewSectionReader(f.r, tgt.off, tgt.sz))
+		dataSize = tgt.h.Size
+	case typ&fs.ModeSymlink != 0: // typ.IsSymlink()
```

Comment:
is it possible `typ.IsRegular() && i.h.Typeflag != tar.TypeLink` and `typ&fs.ModeSymlink != 0` are both true? If so, we should make this the first case, otherwise symlinks won't be handled correctly

Reply:
No, symlinks are not "regular"